### PR TITLE
Integration2 t6 f2 76 rest backend schema

### PR DESCRIPTION
**Pull request by nala-he**
Created Restaurant, Post, FeaturedItem, and Hour objects under the Model directory. Created corresponding Schemas and Models under the Mongoose directory.

### CODE REVIEW

**Review by bovornc — APPROVED** [review]
I think this looks good!

File: models/Restaurant.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    _id?: mongoose.Schema.Types.ObjectId,
+    name: string,
+    bannerPicture?: string,
+    profilePicture?: string,
+    handle: string,
+    bio?: string,
+    cuisine: string,
+    price: string,
+    address: string,
+    // hours?: Hour[],
```
Comment (MaceeQi) [inline models/Restaurant.ts]: Is there a reason these are commented out?

File: models/Restaurant.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * @typedef {Restaurant} Restaurant Represents a restaurant object interface
+ * @property {string} name Restaurant name
+ * @property {string} bannerPicture Banner picture
+ * @property {string} profilePicture Avatar picture
+ * @property {string} handle Restaurant account handle
+ * @property {string} bio Restaurant bio
+ * @property {string} cuisine Restaurant cuisine type
+ * @property {string} price Restaurant price range
+ * @property {string} address Restaurant address
+ * @property {string} hours Restaurant open hours
```
Comment (MaceeQi) [inline models/Restaurant.ts]: JSDoc includes two properties for hours - not sure if you forgot to delete the string hours?

File: mongoose/RestaurantSchema.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @typedef {Restaurant} Restaurant Represents a restaurant object
+ * @property {string} name Restaurant name
+ * @property {string} bannerPicture Banner picture
+ * @property {string} profilePicture Avatar picture
+ * @property {string} handle Restaurant account handle
+ * @property {string} bio Restaurant bio
+ * @property {string} cuisine Restaurant cuisine type
+ * @property {string} price Restaurant price range
+ * @property {string} address Restaurant address
+ * @property {string} hours Restaurant open hours
```
Comment (MaceeQi) [inline mongoose/RestaurantSchema.ts]: string hours listed here in jsdoc but not in actual schema

**Review by MaceeQi — COMMENTED** [review]
A few things I noticed regarding documentation - not sure if they were intentional but just wanted to bring them to your attention. Otherwise, looks good!

File: models/Restaurant.ts (hunk 1 above)
Comment (nala-he) [reply models/Restaurant.ts]: I created them originally and realized that we need the individual objects for Hour, Post, and FeaturedItem. I still kept them there trying to remind myself that these three objects belong to Restaurant. I will delete them.

File: mongoose/RestaurantSchema.ts (hunk 3 above)
Comment (nala-he) [reply mongoose/RestaurantSchema.ts]: Thank you for pointing it out! I deleted it from jsdoc.

File: models/Restaurant.ts (hunk 2 above)
Comment (nala-he) [reply models/Restaurant.ts]: Thank you for pointing it out! I deleted both from jsdoc.

**Comment by nala-he** [comment]
I revised my jsdoc and Restaurant model as suggested by Macee. Thank you for pointing them out!

File: models/Restaurant.ts (hunk 1 above)
Comment (MaceeQi) [reply models/Restaurant.ts]: Okay, no worries if you want to keep them for now! Just wondering why they were there haha

**Comment by MaceeQi** [comment]
Great, thank you Yutong! Looks good to merge : )

**Review by MaceeQi — APPROVED** [review]
Good to merge : )

**Pull request merged**